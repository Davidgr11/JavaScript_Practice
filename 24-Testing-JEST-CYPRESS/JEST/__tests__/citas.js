import Citas from '../js/classes/Citas.js';

describe('Pruebas a la clase Citas', () => {
    const citas = new Citas();
    const id = Date.now()
    test('Agregar una nueva cita', () => {
        const cita = {
            mascota: 'Hook',
            propietario: 'David',
            telefono: '123456789',
            fecha: '2023-10-01',
            hora: '10:00',
            sintomas: 'No come',
            id: id
        };

        citas.agregarCita(cita);
        expect(citas).toMatchSnapshot();
    });

    test('Actualizar una cita existente', () => {
        const citaActualizada = {
            mascota: 'Hook',
            propietario: 'David',
            telefono: '987654321',
            fecha: '2023-10-02',
            hora: '11:00',
            sintomas: 'Vomita',
            id: id
        };

        citas.editarCita(citaActualizada);
        expect(citas).toMatchSnapshot();
    });
    test('Eliminar una cita', () => {
        citas.eliminarCita(id);
        expect(citas).toMatchSnapshot();
    });
});