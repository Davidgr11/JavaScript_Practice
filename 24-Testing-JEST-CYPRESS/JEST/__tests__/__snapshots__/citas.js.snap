// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a la clase Citas Actualizar una cita existente 1`] = `
Citas {
  "citas": [
    {
      "fecha": "2023-10-02",
      "hora": "11:00",
      "id": 1752251209399,
      "mascota": "Hook",
      "propietario": "David",
      "sintomas": "Vomita",
      "telefono": "987654321",
    },
  ],
}
`;

exports[`Pruebas a la clase Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "2023-10-01",
      "hora": "10:00",
      "id": 1752251209399,
      "mascota": "Hook",
      "propietario": "David",
      "sintomas": "No come",
      "telefono": "123456789",
    },
  ],
}
`;

exports[`Pruebas a la clase Citas Eliminar una cita 1`] = `
Citas {
  "citas": [],
}
`;
